modules/imgproc/src/connectedcomponents.cpp

```diff
@@ -1737,14 +1755,40 @@ int cv::connectedComponentsWithStats(InputArray _img, OutputArray _labels, Outpu
     _labels.create(img.size(), CV_MAT_DEPTH(ltype));
     cv::Mat labels = _labels.getMat();
     connectedcomponents::CCStatsOp sop(statsv, centroids);
-    if (ltype == CV_16U){
-        return connectedComponents_sub1(img, labels, connectivity, ccltype, sop);
+    int nbComponents;
+    if (ltype == CV_16U) {
```

Comment:
The first two branches can be joined. Also it's seems that 8U labels are [supported](https://github.com/opencv/opencv/blob/dcbed8d676a4f0879fc25c31aeaf22bd738f2c63/modules/imgproc/src/connectedcomponents.cpp#L1670). Could you check this?